src/test/java/bio/terra/common/stairway/TestMetricExporter.java

```diff
@@ -0,0 +1,40 @@
+package bio.terra.common.stairway;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+
+/** A test metric exporter that stores the last metrics it received. */
+public class TestMetricExporter implements MetricExporter {
+  private Collection<MetricData> lastMetrics;
+
+  @Override
+  public synchronized CompletableResultCode export(Collection<MetricData> metrics) {
+    lastMetrics = Collections.unmodifiableCollection(new ArrayList<>(metrics));
```

Comment:
`List.copyOf()` is a more direct way to create a read-only list that's a copy of an existing colleciton
```suggestion
    lastMetrics = List.copyOf(metrics);
```